llsd/serde_xml.py

```diff
@@ -126,7 +119,46 @@ def _write(self, something):
         """
         self.stream.write(b'<?xml version="1.0" ?>'
                           b'<llsd>')
-        self._generate(something)
+
+        iter_stack = [(iter([something]), b"", None)]
+        while True:
+            cur_iter, iter_type, iterable_obj = iter_stack[-1]
+            try:
+                item = next(cur_iter)
+                if iter_type == b"map":
+
+                    if self.py2: # pragma: no cover
+                        self.stream.write(b'<key>' +
+                                          _str_to_bytes(xml_esc(UnicodeType(item))) +
+                                          b'</key>')
```

Comment:
Consider instead:

    self.stream.writelines((b'<key>',
        _str_to_bytes(xml_esc(UnicodeType(item))),
        b'</key>'))

to avoid having to concatenate in memory before writing in sequence.

Reply:
Fixed in future versions.